contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -8402,6 +8181,86 @@ rewrite_function_call_geospatial_func_ref_args(T ctx)
 // End of Spatial Query Helper for Function Calls 
 ////////////////////////////////////////////////////////////////////////////////
 
+static void
+handleGeospatialFunctionsInFunctionCall(TSqlParser::Function_callContext *ctx)
+{
+	/* Handles rewrite of geospatial function calls */
+	if (ctx->spatial_proc_name_server_database_schema())
+	{
+		if (ctx->spatial_proc_name_server_database_schema()->schema) throw PGErrorWrapperException(ERROR, ERRCODE_FEATURE_NOT_SUPPORTED, "Remote procedure/function reference with 4-part object name is not currently supported in Babelfish", getLineAndPos(ctx));
+
+		/* This if-elseIf clause rewrites the query in case of Geospatial function Call */
+		if (ctx->spatial_proc_name_server_database_schema()->geospatial_func_arg() && ctx->function_arg_list())
+			rewrite_function_call_geospatial_func_ref_args(ctx);
+		else if (ctx->spatial_proc_name_server_database_schema()->geospatial_func_no_arg() && !ctx->function_arg_list())
+			rewrite_function_call_geospatial_func_ref_no_arg(ctx);
+	}
+}
+
+static void 
+handleClrUdtFuncCall(TSqlParser::Clr_udt_func_callContext *ctx)
+{
+	/* checking if CLR_UDT types */
+	if(ctx != NULL && !ctx->DOT().empty())
+	{
+		std::vector<TSqlParser::Method_callContext *> method_calls = ctx->method_call();
+		for (size_t i = 0; i < method_calls.size(); ++i)
+		{
+			TSqlParser::Method_callContext *method = method_calls[i];
+			/* rewriting the query in case of Geospatial function Call */
+			if(method->spatial_methods())
+			{
+				size_t ind = -1;
+				if (i == 0)
+				{
+					if(ctx->local_id()) ind = ctx->local_id()->stop->getStopIndex();
```

Comment:
Why is the test for whitespace is no longer needed?

Reply:
We have directly used that particular index so no need to worry about the spaces in between